openfl/interface/plan.py

```diff
@@ -136,39 +136,40 @@ def initialize(
         # that were not installable via requirements.txt
         plan.get_task_runner(data_loader=None)
     else:
-        init_state_path = plan.config["aggregator"]["settings"]["init_state_path"]
-        # This is needed to bypass data being locally available
-        if input_shape is not None:
-            logger.info(
-                "Attempting to generate initial model weights with custom input shape "
-                f"{input_shape}"
+        aggregator_mode = plan.config["aggregator"]["settings"].get("mode", "learning")
+        if aggregator_mode == "learning":
+            init_state_path = plan.config["aggregator"]["settings"]["init_state_path"]
+            # This is needed to bypass data being locally available
+            if input_shape is not None:
+                logger.info(
+                    f"Attempting to generate initial model weights with custom shape {input_shape}"
+                )
+
+            # Initialize tensor dictionary
+            init_tensor_dict, task_runner, round_number = _initialize_tensor_dict(
+                plan, input_shape, init_model_path
```

Comment:
As I mentioned in a previous commit, we want to avoid too much nesting. Please think of a broader approach that does not create too many edge cases. The goal of adding a `C901` check is to guard the code from getting too complex. By overriding C901 and adding nested conditionals only makes it exceptionally harder to test and validate corner cases